itests/org.openhab.binding.systeminfo.tests/src/main/java/org/openhab/binding/systeminfo/test/SysteminfoOSGiTest.java

```diff
@@ -345,9 +347,15 @@ private void assertItemState(String acceptedItemType, String itemName, String pr
     }
 
     private void intializeItem(ChannelUID channelUID, String itemName, String acceptedItemType) {
+        UnitProvider unitProvider = getService(UnitProvider.class);
```

Comment:
Suggest to make unitProvider a field, similar to itemRegistry and thingRegistry (line 132-133) and move this block to the setUp method (line 190).